### PR TITLE
feat: populate framework over patches

### PR DESCRIPTION
**Pull request by baiorett**

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/ad/infocardsuggestions/patch/HideInfocardSuggestionsPatch.kt [hunk 1]
```diff
@@ -30,6 +33,16 @@ class HideInfocardSuggestionsPatch : BytecodePatch(
     )
 ) {
     override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.ADS.addPreferences(
+            SwitchPreference(
+                "revanced_info_cards_enabled",
+                StringResource("revanced_video_ads_enabled_title", "Show info-cards"),
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/ad/infocardsuggestions/patch/HideInfocardSuggestionsPatch.kt:39]: This has to be renamed

File: src/main/kotlin/app/revanced/patches/youtube/ad/video/patch/VideoAdsPatch.kt [hunk 2]
```diff
@@ -34,10 +34,10 @@ class VideoAdsPatch : BytecodePatch(
         SettingsPatch.PreferenceScreen.ADS.addPreferences(
             SwitchPreference(
                 "revanced_video_ads_enabled",
-                StringResource("revanced_video_ads_enabled_title", "Show video ads"),
+                StringResource("revanced_video_ads_enabled_title", "Hide video ads"),
                 true,
-                StringResource("revanced_video_ads_enabled_summary_on", "Video ads are enabled."),
-                StringResource("revanced_video_ads_enabled_summary_off", "Video ads are disabled.")
+                StringResource("revanced_video_ads_enabled_summary_on", "Video ads are blocked."),
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/ad/video/patch/VideoAdsPatch.kt:39]: Use hidden/shown here

File: src/main/kotlin/app/revanced/patches/youtube/interaction/seekbar/patch/EnableSeekbarTappingPatch.kt [hunk 3]
```diff
@@ -42,7 +55,7 @@ class EnableSeekbarTappingPatch : BytecodePatch(
             if (it.implementation == null) continue
 
             val instructions = it.implementation!!.instructions
-            // here we make sure we actually find the method because it has more then 7 instructions
+            // here we make sure we actually find the method because it has more    than 7 instructions
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/interaction/seekbar/patch/EnableSeekbarTappingPatch.kt:58]: Fix the padding here

File: src/main/kotlin/app/revanced/patches/youtube/misc/autorepeat/patch/AutoRepeatPatch.kt [hunk 4]
```diff
@@ -30,6 +33,16 @@ class AutoRepeatPatch : BytecodePatch(
     )
 ) {
     override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.MISC.addPreferences(
+            SwitchPreference(
+                "revanced_pref_auto_repeat",
+                StringResource("revanced_auto_repeat_enabled_title", "Prefer auto-repeat"),
+                true,
+                StringResource("revanced_auto_repeat_summary_on", "Auto-repeat is preferred."),
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/autorepeat/patch/AutoRepeatPatch.kt:41]: Use "enabled" here

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Review by baiorett — COMMENTED** [review]
done

**Pull request merged**